Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -563,15 +563,15 @@ export class ResourceGroup {
                                     setVariablesInObject(`${path}.${key}`, obj[key]);
                                 }
                                 else {
-                                    tl.setVariable(`${path}.${key}`, JSON.stringify(obj[key]));
+                                    console.log(`##vso[task.setvariable variable=${path}.${key};]` + JSON.stringify(obj[key]));
```

Comment:
This will print it in the task logs. Does `tl.setVariable` also do that? Not quite sure, but I think no.

The output to a template deployment can contain a secret. I believe there is a property called `isSecret` which can be used here.

Reply:
`${path}.${key}` means that key will contain a dot (`.`). Please check that it does not break the ## command

Reply:
No, it doesn't print  unless debug is off
https://dev.azure.com/vineetpc/vnm-arg-dts-03-2021/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?_a=release-environment-logs&releaseId=5&environmentId=5

Reply:
> ${path}.${key} means that key will contain a dot (.). Please check that it does not break the ## command

it doesn't

Reply:
> The output to a template deployment can contain a secret. I believe there is a property called isSecret which can be used here.

isSecret wasn't used in  `tl.setVariable(`${path}.${key}`, JSON.stringify(obj[key]));`, why to use now?